packages/carbon-component-mapper/src/tests/text-field.test.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { act } from 'react-dom/test-utils';
+
+import { FormRenderer, componentTypes } from '@data-driven-forms/react-form-renderer';
+
+import FormTemplate from '../form-template';
+import componentMapper from '../component-mapper';
+import { TextInput, NumberInput } from 'carbon-components-react';
+
+describe('<TextInput and NumberInput />', () => {
+  let onChange;
+  let wrapper;
+  beforeEach(() => {
+    onChange = jest.fn();
+  });
+  afterEach(() => {
+    onChange.mockReset();
+  });
+  it('renders NumberInput', () => {
+    const schema = {
+      fields: [
+        {
+          component: componentTypes.TEXT_FIELD,
+          name: 'input',
+          label: 'Please enter a value',
+          type: 'number',
+          step: 1,
+          initialValue: 1,
+          onChange
+        }
+      ]
+    };
+
+    const wrapper = mount(
+      <FormRenderer onSubmit={jest.fn()} FormTemplate={(props) => <FormTemplate {...props} />} schema={schema} componentMapper={componentMapper} />
+    );
+
+    expect(wrapper.find(NumberInput)).toHaveLength(1);
+  });
+
+  it('NumberInput - click on increment button', async () => {
+    const schema = {
+      fields: [
+        {
+          component: componentTypes.TEXT_FIELD,
+          name: 'input',
+          label: 'Please enter a value',
+          type: 'number',
+          step: 1,
+          initialValue: 1,
+          onChange
+        }
+      ]
+    };
+    await act(async () => {
+      wrapper = mount(
+        <FormRenderer onSubmit={jest.fn()} FormTemplate={(props) => <FormTemplate {...props} />} schema={schema} componentMapper={componentMapper} />
+      );
+    });
+
+    expect(wrapper.find(NumberInput)).toHaveLength(1);
+    expect(wrapper.find('input')).toHaveLength(1);
+
+    await act(async () => {
+      wrapper
+        .find('button.up-icon')
+        .first()
+        .simulate('click');
+    });
+    wrapper.update();
+
+    const valueInput = wrapper.find('input[type="number"]');
+    expect(valueInput).toHaveLength(1);
+    expect(valueInput.props().value).toEqual(2);
```

Comment:
We should test form state, not the value prop - the value prop can came up from the carbon component, so it can just “look” ok.

You can check example [here](https://data-driven-forms.org/testing#testingtherenderer). Just put a spy function to renderer submit and submit the form via `find(“form”).simulate(“submit”)` in `act` and check what values are received in the first argument.

Reply:
@rvsia i tried it write it the way you mentioned, i am getting error,  not really sure what i missed, any idea?
<img width="623" alt="Screen Shot 2021-04-19 at 2 13 21 PM" src="https://user-images.githubusercontent.com/37085529/115283518-7985d380-a119-11eb-8fbe-8f344e3aa071.png">

<img width="541" alt="Screen Shot 2021-04-19 at 2 13 26 PM" src="https://user-images.githubusercontent.com/37085529/115283519-7985d380-a119-11eb-80bc-1a5b9bffff94.png">

<img width="1114" alt="Screen Shot 2021-04-19 at 2 14 22 PM" src="https://user-images.githubusercontent.com/37085529/115283588-915d5780-a119-11eb-8b44-187754cc0921.png">

Reply:
Did you pass the spy function to renderer?

Reply:
@rvsia your guess is correct, i haven't passed that :). updated it,  now getting new error
<img width="1326" alt="Screen Shot 2021-04-19 at 2 31 21 PM" src="https://user-images.githubusercontent.com/37085529/115285568-0467cd80-a11c-11eb-8357-05d3b67297d8.png">

Reply:
This is correct, there is formApi as the second argument in the submit function and some callback as the third. You don’t have to check these ones.

You can pick just the values in the submit prop:

`onSubmit={values => onSubmit(values)}`

Reply:
@rvsia ah ok makes sense, i updated it now, i am still having issue with expected values, i am simulating clicks on increment and decrement buttons here, but on submit, it is still giving out initial value, i pushed my changes , please check.
Note: code i wrote for value check , it is working as expected.

Reply:
Yeah, it’s important to send the correct values in the submit.. I will check tomorrow morning what is going on, getting sleepy. 😴 Thank you for now! 👍

Reply:
Cool, thank you.